src/form/input/number/util/numberInputUtils.ts

```diff
@@ -0,0 +1,205 @@
+    : {
+        locale: undefined,
+        shouldFormatToLocaleString: false,
+        DECIMAL_NUMBER_SEPARATOR: undefined,
+        MINUS_SIGN: undefined,
+        LOCALE_NEGATIVE_ZERO: undefined
+      };
+}
+
+function getNumberInputParseNumberOptions(
+  options: ParseNumberOptions
+): ParseNumberOptions {
+  return shouldActiveNumberInputLocalization()
+    ? options
+    : {
+        ...options,
+        locale: "en-US"
+      };
+}
+
+function isNumberInputValueNotAValidNumber({
+  eventValue,
+  formattedValue
+}: {
+  eventValue: string;
+  formattedValue: string;
+}) {
+  return (
+    formattedValue === "" || (eventValue !== "-" && Number.isNaN(Number(formattedValue)))
+  );
+}
+
+function getIsValidNumberInputMaxFractionDigits({
+  maximumFractionDigits
+}: {
+  maximumFractionDigits: number;
+}) {
+  return Number.isInteger(maximumFractionDigits) && maximumFractionDigits >= 0;
+}
+
+function localizeNumberInputValue({
+  value,
+  formatProps,
+  maximumFractionDigits
+}: {
+  value: React.InputHTMLAttributes<HTMLInputElement>["value"];
+  formatProps: NumberInputFormatProps;
+  maximumFractionDigits: number;
+}) {
+  const {
+    locale,
+    shouldFormatToLocaleString,
+    DECIMAL_NUMBER_SEPARATOR: decimalSeparatorForLocale,
+    MINUS_SIGN: minusSignForLocale,
+    LOCALE_NEGATIVE_ZERO: negativeZeroForLocale
+  } = getNumberInputFormatProps(formatProps);
+  let finalValue = value;
+
+  if (typeof value === "string" && shouldFormatToLocaleString) {
+    const [integerPart, decimalPart] = String(value).split(".");
+    const numberFormatter = formatNumber({
+      maximumFractionDigits,
+      locale
+    });
+
+    // IF there is a decimal part or the value ends with ".",
+    // make sure we add the decimal separator and map each digit on the decimal part to localized versions.
+    // We shouldn't use parseInt or parseFloat with numberFormat util here because that removes zeros on the decimal part and disallows users to write something like: 10.01 or 10.102
+    if (value.match(/\.$/)?.length || decimalPart) {
+      finalValue = `${numberFormatter(
+        parseInt(integerPart)
+      )}${decimalSeparatorForLocale}${mapDigitsToLocalVersion({locale}, decimalPart)}`;
+    } else if (integerPart) {
+      if (integerPart !== minusSignForLocale && integerPart !== negativeZeroForLocale) {
+        finalValue = numberFormatter(parseInt(integerPart));
+      } else {
+        finalValue = `${minusSignForLocale}${mapDigitsToLocalVersion(
+          {locale},
+          integerPart
+        )}`;
+      }
+    }
+  }
+
+  return finalValue;
+}
+
+function delocalizeNumberInputValue({
+  value,
+  event,
+  formatProps,
+  maximumFractionDigits
+}: {
+  value: React.InputHTMLAttributes<HTMLInputElement>["value"];
+  event: React.SyntheticEvent<HTMLInputElement>;
+  formatProps: NumberInputFormatProps;
+  maximumFractionDigits: number;
+}) {
+  const {locale, shouldFormatToLocaleString} = getNumberInputFormatProps(formatProps);
```

Comment:
`getNumberInputFormatProps` and `getNumberInputParseNumberOptions` can be used directly within the component body and their returned values can be provided to the helper utility functions as arguments instead